JuiceMaker/JuiceMaker/Controller/JuiceMakerViewController.swift

```diff
@@ -11,40 +11,39 @@ class JuiceMakerViewController: UIViewController {
         super.viewDidLoad()
         
         updateAllFruitsCount()
+        
+        NotificationCenter.default.addObserver(
+            self,
+            selector: #selector(updateFruitCount),
+            name: Notification.Name("showFruitCount"),
+            object: nil)
     }
     
-    private func requestUpdateFruitCount(fruitJuice: FruitJuice) {
-        switch fruitJuice.ingridientCount {
-        case 1:
-            let (fruit, _) = fruitJuice.juiceIngridients.first
-            updateFruitCount(fruit: fruit)
-        case 2:
-            let (fruit1, _) = fruitJuice.juiceIngridients.first
-            guard let (fruit2, _) = fruitJuice.juiceIngridients.second else { return }
-            updateFruitCount(fruit: fruit1)
-            updateFruitCount(fruit: fruit2)
-        default:
-            return
+    @objc func updateFruitCount(_ notification: Notification) {
+        let currentFruitList: [Fruits: String?] = [
+            .strawberry: strawberryCountLabel.text,
+            .banana: bananaCountLabel.text,
+            .pineapple: pineappleCountLabel.text,
+            .kiwi: kiwiCountLabel.text,
+            .mango: mangoCountLabel.text
+        ]
+        
+        guard let changedFruitList = notification.object as? [Fruits: Int] else { return }
+        
+        for (fruit, count) in changedFruitList {
+            if currentFruitList[fruit] != String(count) {
+                updateCountLabel(fruit: fruit, count: count)
+            }
         }
     }
```

Comment:
```suggestion
         for (fruit, count) in changedFruitList where currentFruitList[fruit] != String(count) {
            updateCountLabel(fruit: fruit, count: count)
        }
```

함수 내 깊이를 줄이는 것 항상 고민해주세요!

Reply:
😮for in에 where를 붙여서 저런식으로 써줄수있군요! 하나 알아갑니다 ㅎㅎ